pyfr/plugins/writer.py

```diff
@@ -58,6 +58,18 @@ def __call__(self, intg):
         else:
             metadata = None
 
+        # Fetch data from other plugins and add it to metadata with ad-hoc keys
+        for csh in intg.completed_step_handlers:
+            try:
+                opdata = csh.serialise(intg)
+
+                if rank == root:
+                    prefix = intg.get_plugin_data_prefix(csh.name, csh.suffix)
+                    metadata.update(
+                        {f'{prefix}/{k}': v for k, v in opdata.items()})
+            except AttributeError:
```

Comment:
Move the except up and then in the failover block do `opdata = {}` this ensures that our try/except block is as small as possible.